### ISSUE TITLE
Enhance Coverage of Distributed Systems Fundamentals

### ISSUE THREAD

**Issue by ZCH-SA** [isaqb-org/curriculum-foundation#681]
**Overview**
Current coverage of distributed systems concepts in LG 03-12 needs enhancement to better prepare architects for addressing common distributed systems challenges while maintaining foundation-level appropriateness.

**Current State**
LG 03-12 currently provides basic coverage of distributed systems concepts, focusing on identifying distribution, analyzing consistency criteria, and explaining causality. The treatment of practical patterns and trade-offs is limited.

**Desired State**
Foundation-level content that enables participants to understand and make informed decisions about distributed systems architecture, with clear examples and practical guidance on common patterns and challenges.

**Proposed Changes**

- Add practical examples of basic distributed system patterns
- Provide foundational understanding of consistency vs. availability choices
- Include basic patterns for handling distribution challenges
- Enhance coverage of basic microservices patterns from LG 03-08

**Additional Context**
Changes should maintain foundation-level focus while providing enough practical guidance for participants to understand and apply these concepts in real-world situations.

**Comment by mahboubagharbi** [comment]
I suspect the syllabus is getting really advanced 😅. We should differentiate a little here between what belongs in a basic course and what belongs in an advanced module.

**Comment by ZCH-SA** [comment]
This is a great point, and I completely agree on the importance of maintaining the foundational focus. Distributed systems, however, are increasingly becoming a core part of even medium-sized projects, which the foundation level aims to address. Providing architects with a basic understanding of key concepts like consistency trade-offs or practical patterns (e.g., message queues or service discovery) ensures they can make informed, grounded decisions without stepping ß0into advanced complexities.

The intention here is not to expand into advanced territory but to refine how we present these essentials—keeping it practical, approachable, and directly relevant to the challenges architects are likely to face in their early projects. By introducing these concepts with clear examples and minimal theoretical overhead, we can empower participants to avoid common pitfalls while sticking to the course's foundational scope.

**Comment by rhoadesre** [comment]
I agree with Mahbouba. This is encroaching too much on advanced topics that should be covered in advanced classes like Flex and Cloud.

**Comment by rhoadesre** [comment]
@Zied: it would be helpful if you could add concrete proposals for text passages in the curriculum.

**Comment by mikesperber** [comment]
> This is encroaching too much on advanced topics that should be covered in advanced classes like Flex and Cloud.

First of all, this is not presently the case - CLOUDINFRA mentions distributed systems in passing only, and FLEX lists them as prerequisites.  Also, as mentioned, distributed systems are pervasive, and it is hard to see how they're not relevant for everybody doing software architecture. 

So really, this needs to be mandatory material on the iSAQB certification path. At the same time, it's clear that we don't have the space and time in Foundation. It's a real problem.

**Comment by mahboubagharbi** [comment]
Not really a problem; however, let's address it as an issue for the FLEX curriculum.

**Comment by aheusingfeld** [comment]
> and FLEX lists them as prerequisites

Well, I just re-read the prerequisites section of the FLEX curriculum. First of all, there are no MUST criteria in the list: https://public.isaqb.org/curriculum-flex/curriculum-flex-en.html#_prerequisites

We state that experience with "Problems and challenges in the implementation of distributed systems" are useful for understanding some concepts. That is, if you already experienced a problem, the trainer doesn't have to explain in broad detail anymore but you can relate to why you want to have a solution for this.

**Comment by gernotstarke** [comment]
to keep our own Foundation repo clean, I close this issue, as it needs to be handled elsewhere. @aheusingfeld and @mahboubagharbi are involved in corresponding topics.
